test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,291 @@
+/*
```

Comment:
This isn't really a roundtrip test right now.  Are you going to add round-tripping?